src/js/game/LevelMVC/LevelPlane.js

```diff
@@ -325,9 +325,148 @@ module.exports = class LevelPlane extends Array {
+          let blocksPositions = this.getBlocksToPush(neighborPosition[1], offset);
+          this.pushBlocks(blocksPositions, offset);
+        } else if (neighbors.south.block.blockType === "") {
+          let armBlock = new LevelBlock("pistonArm");
+          this.setBlockAt(neighborPosition[1], armBlock);
+        }
+        break;
+      }
+      case "pistonRight": {
+        if (neighbors.east.block.blockType !== "" && neighbors.east.block.blockType !== "pistonArm") {
+          let offset = [1,0];
+          let blocksPositions = this.getBlocksToPush(neighborPosition[2], offset);
+          this.pushBlocks(blocksPositions, offset);
+        } else if (neighbors.east.block.blockType === "") {
+          let armBlock = new LevelBlock("pistonArm");
+          this.setBlockAt(neighborPosition[2], armBlock);
+        }
+        break;
+      }
+      case "pistonLeft": {
+        if (neighbors.west.block.blockType !== "" && neighbors.west.block.blockType !== "pistonArm") {
+          let offset = [-1,0];
+          let blocksPositions = this.getBlocksToPush(neighborPosition[3], offset);
+          this.pushBlocks(blocksPositions, offset);
+        } else if (neighbors.west.block.blockType === "") {
+          let armBlock = new LevelBlock("pistonArm");
+          this.setBlockAt(neighborPosition[3], armBlock);
+        }
+        break;
+      }
+    }
+  }
+
+  deactivatePiston(position) {
+    let neighborPosition = this.getOrthogonalPositions(position);
+    let emptyBlock = new LevelBlock("");
+
+    switch (this[this.coordinatesToIndex(position)].blockType) {
+      case "pistonUp": {
+        if (this[this.coordinatesToIndex(neighborPosition[0])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[0], emptyBlock, false, true);
+        }
+        break;
+      }
+      case "pistonDown": {
+        if (this[this.coordinatesToIndex(neighborPosition[1])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[1], emptyBlock, false, true);
+        }
+        break;
+      }
+      case "pistonRight": {
+        if (this[this.coordinatesToIndex(neighborPosition[2])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[2], emptyBlock, false, true);
+        }
+        break;
+      }
+      case "pistonLeft": {
+        if (this[this.coordinatesToIndex(neighborPosition[3])].blockType === "pistonArm") {
+          this.setBlockAt(neighborPosition[3], emptyBlock, false, true);
+        }
+        break;
+      }
+    }
+  }
+
+  pushBlocks(blocksPositions, offset) {
+    let armBlock = new LevelBlock("pistonArm");
+    // Need to check if the last index is up against the edge of the map, and break out if that's the case.
+    if (!this.inBounds([blocksPositions[blocksPositions.length - 1][0] + offset[0], blocksPositions[blocksPositions.length - 1][1] + offset[1]])) {
+      // Apparently me might NOT want this check? Or might? Design undecided.
+      //break;
+    }
```

Comment:
Not used?

Reply:
Ah, yeah. I didn't know exactly what to do with that. Let me go see if Design input can still be found.

Reply:
Confirmed taking this out.